### PR TITLE
tests: SegmentBitmapFilterTest supports common handle

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #9963

### What is changed and how it works?

- Make `SegmentBitmapFilterTest` support common handle.
- Introduce parameter `including_right_boundary` in some test utils to support  generating test data with `std::numeric_limits<Int64>::max()`.
  - The max value of a left-closed and right-open intervals, such as `[a, std::numeric_limits<Int64>::max())`, is `std::numeric_limits<Int64>::max() - 1`. This results in the inability to test certain boundary conditions.
  - If `including_right_boundary` is true, `[a, b)` turns into `[a, b]`.

```commit-message
- Make `SegmentBitmapFilterTest` support common handle.
- Introduce parameter `including_right_boundary` in some test utils to support generating test data with `std::numeric_limits<Int64>::max()`.
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by Lloyd-Pottiger** [comment]
`including_right_boundary` is not necessary, just use closed interval is enough. [a, b) -> [a, b - 1], [a, +inf) -> [a, MAX]

**Review by Lloyd-Pottiger — APPROVED** [review]
lgtm

File: dbms/src/Storages/DeltaMerge/tests/gtest_segment_util.cpp [hunk 1]
```diff
@@ -21,58 +21,102 @@ namespace DB::DM::tests
...
+    for (auto & r : results)
+        boost::trim(r);
+    return results;
+}
+
+// [a, b) => {a, b, false}
+// [a, b] => {a, b, true}
 template <typename T>
-std::pair<T, T> parseRange(String & str_range)
+std::tuple<T, T, bool> parseRange(std::string_view s)
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/tests/gtest_segment_util.cpp:40]: What about use a named struct `Range<T>` instead of `std::tuple<T, T, bool>`?

File: dbms/src/Storages/DeltaMerge/tests/gtest_segment_util.h [hunk 2]
```diff
@@ -23,15 +24,35 @@ namespace DB::DM::tests
 struct SegDataUnit
 {
     String type;
-    std::pair<Int64, Int64> range; // Data range
+    // {left, right, including_right_boundary}
+    // `including_right_boundary` is required if we want to generate data with int64_max.
+    std::tuple<Int64, Int64, bool> range;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/tests/gtest_segment_util.h:29]: ditto

**Review by JaySon-Huang — COMMENTED** [review]
Basically LGTM with a nit comment

**Review by JaySon-Huang — APPROVED** [review]
